internal/armada/server/submit.go

```diff
@@ -529,6 +534,16 @@ func (server *SubmitServer) createJobs(request *api.JobSubmitRequest, owner stri
 	return jobs, nil
 }
 
+func enrichText(text map[string]string, jobId string) map[string]string {
+	returnText := util.DeepCopy(text)
+
+	for key, value := range returnText {
+		returnText[key] = strings.Replace(value, "{JobId}", jobId, -1)
```

Comment:
although unlikely there is no way to escape `{JobId}`. I think we should provide a way to escape this value incase users want to use it literally.

Reply:
I don't believe {} are allowed in labels or annotations anyway are they?

Reply:
I've not tested but I suspect almost anything will be allowed in annotation values.

Reply:
I'm still not entirely convinced this isn't just a YAGNI, if you have any concrete example of when someone would want to have `{JobId}` in their annotation then I'll add it but it seems more than feasible they could use `{Job_Id}` or some other variation instead. What is the use case for having the escaped string there anyway?

Reply:
Annotations by their nature are very flexible and it's just good practice as the platform to not get in the way. Escaping the magic token surely can't add much complexity to the code and providing a way to escape it will be expected and intuitive to users.